File: solutions/go/02/grid_4dir/main.go
package main

import "fmt"

func main() {
	grid := [][]byte{
		[]byte("..#."),
		[]byte(".#.."),
		[]byte("...."),
	}
	h, w := len(grid), len(grid[0])
	di := []int{-1, 1, 0, 0}
	dj := []int{0, 0, -1, 1}
	si, sj := 1, 0
	var ok [][2]int
	for k := 0; k < 4; k++ {
		ni, nj := si+di[k], sj+dj[k]
		if 0 <= ni && ni < h && 0 <= nj && nj < w && grid[ni][nj] != '#' {
			ok = append(ok, [2]int{ni, nj})
		}
	}
	fmt.Println(ok)
}

File: solutions/python/02/grid_4dir/main.py
import sys


def main() -> None:
    grid = [
        list("..#."),
        list(".#.."),
        list("...."),
    ]
    h, w = len(grid), len(grid[0])
    di = [-1, 1, 0, 0]
    dj = [0, 0, -1, 1]
    si, sj = 1, 0
    ok = []
    for k in range(4):
        ni, nj = si + di[k], sj + dj[k]
        if 0 <= ni < h and 0 <= nj < w and grid[ni][nj] != "#":
            ok.append((ni, nj))
    print(ok)


if __name__ == "__main__":
    main()
